packages/eventual-send/src/local.js

```diff
@@ -13,6 +18,55 @@ const ntypeof = specimen => (specimen === null ? 'null' : typeof specimen);
  */
 const isObject = val => Object(val) === val;
 
+const BREAKPOINTS = getEnvironmentOption('BREAKPOINTS', '');
+
+/**
+ * @type {Record<string, [string, (number | '*')][]>}
+ */
+// @ts-expect-error doesn't like null __proto__
+const BREAKS = { __proto__: null };
+
+const BP_REGEXP = /(.+?)\.(\w+?|\*)=(\d+?|\*)/;
+
+for (const bp of BREAKPOINTS.split(',')) {
+  if (bp !== '') {
+    const match = BP_REGEXP.exec(bp);
+    if (match === null) {
+      throw Fail`Expected "tagName.methodName=count" ${q(bp)}`;
+    }
+    const [_, tag, verb, countStr] = match;
+    const patts = hasOwn(BREAKS, verb) ? BREAKS[verb] : (BREAKS[verb] = []);
+    patts.push([tag, countStr === '*' ? countStr : +countStr]);
+  }
+}
```

Comment:
Grouping primarily by method name/wildcard in appearance order and secondarily by object tag/wildcard in appearance order implies a particular specificity resolution such that e.g. `BREAKPOINTS=Alleged: Foo.*=1,Alleged: Foo.m=2,*.m=3,Alleged: Bar.m=4,*.m2=5` will result in `E(fooInstance).m()` using the 2 while `E(barInstance).m()` and `E(bazInstance).m()` use the 3 (which in particular completely supersedes the 4), and `E(fooInstance).m2()` using the 5. Prioritizing method name over object tag seems fine, but we probably want wildcards to be considered last in each group. For example:
```suggestion
    // eslint-disable-next-line no-nested-ternary
    const count = countStr === '*'
      ? countStr
      : countStr === '0'
        ? 0
        : (+countStr || Fail`invalid skip count in delivery breakpoint filter ${q(bp)}`);
    const patts = hasOwn(BREAKS, verb) ? BREAKS[verb] : (BREAKS[verb] = []);
    const i = tag === '*' ? -1 : patts.findIndex(patt => patt[0] === '*');
    if (i === -1) {
      patts.push([tag, count]);
    } else {
      patts.splice(i, 0, [tag, count]);
    }
  }
}
if (BREAKS['*']) {
  const wildcard = BREAKS['*'];
  delete BREAKS['*'];
  BREAKS['*'] = wildcard;
}
```

Reply:
@gibson042 this internal representation is now encapsulated but is not yet reformed. So, not yet resolving, but noting to clarify further review.

@dckc , I refactored the decision about whether to breakpoint into a separate predicate function that does not itself do the breakpointing, but instead leaves that to its caller. As a result, I was able to write unit tests testing that predicate. PTAL.

Reply:
With the new nested-record internal representation, I was more easily able to consistently always check the '*' cases last. Resolving.